ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/utils.py

```diff
@@ -184,3 +188,108 @@ def _get_l3_interface_cfg(self, l3_interface: dict) -> dict | None:
             interface["flow_tracker"] = {"hardware": self.shared_utils.wan_flow_tracker_name}
 
         return strip_empties_from_dict(interface)
+
+    def _get_l3_uplink_with_l2_as_subint(self, link: dict) -> (dict, list[dict]):
+        """
+        Return a tuple with main uplink interface, list of subinterfaces representing each SVI.
+        """
+        vlans = [int(vlan) for vlan in range_expand(link["vlans"])]
+
+        # Main interface
+        # Routed interface with no config unless there is an SVI matching the native-vlan, then it will contain the config for that SVI
+
+        interfaces = []
+        for tenant in self.shared_utils.filtered_tenants:
+            for vrf in tenant["vrfs"]:
+                for svi in vrf["svis"]:
+                    svi_id = int(svi["id"])
+                    # Skip any vlans not part of the link
+                    if svi_id not in vlans:
+                        continue
+
+                    interfaces.append(self._get_l2_as_subint(link, svi, vrf))
+
+        # If we have the main interface covered, we can just exclude it from the list and return as main interface.
+        # Otherwise we return an almost empty dict as the main interface since it was already covered by the calling function.
+        main_interface = get_item(interfaces, "name", link["interface"], default={"type": "routed", "mtu": self.shared_utils.p2p_uplinks_mtu})
+        main_interface.pop("description", None)
+
+        if (mtu := main_interface.get("mtu", 1500)) != self.shared_utils.p2p_uplinks_mtu:
+            raise AristaAvdError(
+                f"MTU '{self.shared_utils.p2p_uplinks_mtu}' set for 'p2p_uplinks_mtu' conflicts with MTU '{mtu}' "
+                f"set on SVI for uplink_native_vlan '{link['native_vlan']}'."
+                "Either adjust the MTU on the SVI or p2p_uplinks_mtu or change/remove the uplink_native_vlan setting."
+            )
+        return main_interface, [interface for interface in interfaces if interface["name"] != link["interface"]]
+
+    def _get_l2_as_subint(self, link: dict, svi: dict, vrf: dict) -> dict:
+        """
+        Return structured config for one subinterface representing the given SVI.
+        Only supports static IPs or VRRP.
+        """
+        svi_id = int(svi["id"])
```

Comment:
we do this in two places (line 204 and here)

Reply:
we do it to a local variable in line 204, but that local variable is not given to this function, so we have to do it again.